numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -2365,9 +2363,167 @@ static PySequenceMethods voidtype_as_sequence = {
 };
 
 
-static PyBufferProcs gentype_as_buffer = {
-    .bf_getbuffer = gentype_getbuffer,
-    /* release buffer not defined (see buffer.c) */
+
+/**begin repeat
+ * #name = bool, byte, short, int, long, longlong, ubyte, ushort, uint, ulong,
+ *         ulonglong, half, float, double, longdouble, cfloat, cdouble,
+ *         clongdouble#
+ * #Name = Bool, Byte, Short, Int, Long, LongLong, UByte, UShort, UInt, ULong,
+ *         ULongLong, Half, Float, Double, LongDouble, CFloat, CDouble,
+ *         CLongDouble#
+ * #NAME = BOOL, BYTE, SHORT, INT, LONG, LONGLONG, UBYTE, USHORT, UINT, ULONG,
+ *         ULONGLONG, HALF, FLOAT, DOUBLE, LONGDOUBLE, CFLOAT, CDOUBLE,
+ *         CLONGDOUBLE#
+ * #fmt = ?, b, h, i, l, q, B, H, I, L, Q, e, f, d, g, Zf, Zd, Zg#
```

Comment:
Does this do the right thing on platforms with unusually-sized ints?

Reply:
I think the tests here should actually be pretty good:

https://github.com/numpy/numpy/blob/d0b7b6638fe7496d25a488a179d79879748075fa/numpy/core/tests/test_scalarbuffer.py#L9-L30

Reply:
I need to reread the buffer protocol I think. My fear is platforms that we don't test on, like those with 16-bit char.

Reply:
The buffer protocol uses the struct definitions: https://docs.python.org/3.8/library/struct.html

Reply:
Looks like my concern was only for non-native formats, but scalars are always fixed endianness so this is safe